jdaviz/configs/imviz/plugins/reproject/reproject.py

```
@@ -0,0 +1,122 @@
+
+try:
+    from reproject import reproject_interp
+    from reproject.mosaicking import find_optimal_celestial_wcs
+except ImportError:
+    HAS_REPROJECT = False
+else:
+    HAS_REPROJECT = True
+
+__all__ = ['Reproject']
+
+
+@tray_registry('imviz-reproject', label="Reproject")
+class Reproject(PluginTemplateMixin, DatasetSelectMixin, AutoTextFieldMixin):
+    template_file = __file__, "reproject.vue"
+
+    reproject_in_progress = Bool(False).tag(sync=True)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        if not HAS_REPROJECT:
+            self.disabled_msg = 'Please install reproject and restart Jdaviz to use this plugin'
+            return
+
+    @property
+    def user_api(self):
+        return PluginUserApi(self, expose=('dataset', 'label', 'reproject'))
+
+    @observe("dataset_selected")
+    def _set_default_results_label(self, event={}):
+        '''Generate a label and set the results field to that value'''
+        self.label_default = f"{self.dataset_selected} (reprojected)"
+
+    @observe('label')
+    def _label_changed(self, event={}):
+        if not len(self.label.strip()):
+            # strip will raise the same error for a label of all spaces
+            self.label_invalid_msg = 'label must be provided'
+            return
+        if self.label.strip() in self.data_collection:
+            self.label_invalid_msg = 'label already in use'
+            return
+        self.label_invalid_msg = ''
+
+    def reproject(self):
+        """
+        Reproject ``dataset`` so that North is up in a new entry labeled ``label`` and set as the
+        reference image.
+        """
+        if (not HAS_REPROJECT or self.dataset_selected not in self.data_collection
+                or self.reproject_in_progress):
+            return
+
+        import reproject
+
+        data = self.data_collection[self.dataset_selected]
+        wcs_in = data.coords
+        if wcs_in is None:
+            return
```

should this raise an error or snackbar instead of failing silently?

---

Added a snackbar message.